### DISCUSSION TITLE
how to customize PVD1 and ESD1 models and link of reference diagrams

### DISCUSSION THREAD

**Discussion by SanjeevKumarGoyal** [CURENT/andes#314] [Q&A]
Good morning Dr. Cui,
I am working on PVs and Storage (Batter) and using `PVD1`, `ESD1 `input models for the same.
I further want to edit and modify the **input model** and want to customize it.
Is it possible to add **Power Factor control** to any of existing model as mentioned above?

Please provide the reference diagram for above mentioned models i.e. `PVD1`, `ESD1`

below link represents the **controller** for these models, please clarify.
https://www.powerworld.com/WebHelp/Content/TransientModels_HTML/Machine%20Model%20PVD1.htm

Regards,
Sanjeev

**Comment by cuihantao** [comment]
Hi,

You can check the code for REECA1 and borrow some from there.

Regards,
Hantao Cui

**Comment by JunejaAmmy** [comment]
Hi Dr. Cui,

Could you please provide the reference diagrams for models `PVD1 `and `ESD1`?

And please clarify if the below diagram represents the controller for PVD1 model.
https://www.powerworld.com/WebHelp/Content/TransientModels_HTML/Machine%20Model%20PVD1.htm

**Reply by cuihantao** [reply]
The link you posted is the diagram for PVD1.

We don't have a drawing for ESD1 yet. It is modified from PVD1, and you can read the code to quickly figure out the addition. You are welcome to contribute a diagram.

**Reply by SanjeevKumarGoyal** [reply]
Good Morning Dr. Cui,

In Manual, below is the link provided under `ESD1`:
https://www.powerworld.com/WebHelp/Content/TransientModels_HTML/Exciter%20REEC_C.htm

Please have a look and verify about the same.

Regards,
Sanjeev

**Reply by cuihantao** [reply]
The document is clear enough. You can find the state-of-charge control loop, which is added to PVD1 to form ESD1, in the REEC_C model. What do you want me to verify? Why don't you check the diagram and check the code? There is no point to keep asking questions if you have not done the homework.
